docs/src/tutorials/MHMPC.md

```diff
@@ -0,0 +1,207 @@
+
+R = Diagonal([1.0])  # input cost matrix 
+Q = Diagonal([1.0, 1.0])
+
+NX = 2 # Number of x variables [position, velocity]
+NU = 1 # Number of u variables [force]
+```
+
+### 2. Make graph structure
+
+We define our graph and subgraph structure. This encapsulates the MH-MPC approach for each $H$ sub-interval  $i \in H$. Here $H = 5$
+
+```julia
+graph_master = OptiGraph() # This will contain all subgraphs
+
+# All of the subgraphs
+graph_SG1 = OptiGraph()
+graph_SG2 = OptiGraph()
+graph_SG3 = OptiGraph()
+graph_SG4 = OptiGraph()
+graph_SG5 = OptiGraph()
+
+subgraphs_all = [graph_SG1, graph_SG2, graph_SG3, graph_SG4, graph_SG5]
+
+# Initialize all nodes for subgraphs
+@optinode(graph_SG1, nodes1[1:N_i[1]])
+@optinode(graph_SG2, nodes2[1:N_i[2]])
+@optinode(graph_SG3, nodes3[1:N_i[3]])
+@optinode(graph_SG4, nodes4[1:N_i[4]])
+@optinode(graph_SG5, nodes5[1:N_i[5]])
+
+nodes_all = [graph_SG1[:nodes1], graph_SG2[:nodes2], graph_SG3[:nodes3], graph_SG4[:nodes4], graph_SG5[:nodes5]]
+```
+
+The second parameter of ``@optinode`` will initialize $N_i$ nodes. They can be referenced (as seen later) by calling the name and the subsequent index of that specific node (e.g. ``nodes2[1]`` for the first node of the second subgraph.). The arrays ``subgraphs_all`` and ``nodes_all`` are used for easier, ordered indexing.
+
+### 3. Initialize variables for each node
+
+This step is necessary for initializing the state and input variables. Recall, each node is representative of a time point in the MH-MPC problem. ``@variable`` is a macro to initialize a variable for the specific node in the first parameter. The constraints for $x$ and $u$ are implemented using the inequality constraints seen in the second parameter. Additionally, the vector notation for $x,u$ will initialize each variable as a vector of size $NX, NU$, respectively.
+
+``@expression`` is another macro that defines a specific symbolic expression for each node. This expression will add simplifications for the following code representative of the first equality constraint: $x_{k+1} = A_k x_k + B_k x_k$. This expression is named via the second parameter.
+
+Additionally, the initial condition for this problem is defined for $t=0$, where $x_0 = [18\quad \mathord{-} 14]$. The ``fix`` function will fix a specified variable. An important note about variable referencing, ` nodes1[1][:x][1]` has a lot going on.
+
+```julia
+for (subgraph, t_step) in zip(subgraphs_all, t_i)
+    # Instantiate x,u for each node (time point)
+    for node in all_nodes(subgraph)
+        @variable(node, -20 <= x[1:NX] <= 20)
+        @variable(node, -0.5 <= u[1:NU]<= 0.5)
+
+        @variable(node, dt == t_step)
+
+        A = [1 dt; -dt (1-dt/2)]
+        B = [0 ; 2*dt]
+
+        @expression(node, Ax_Bu[i = 1:NX],  A[i, 1] * node[:x][1] 
+        + A[i, 2] * node[:x][2] + B[i] * node[:u][1])
+    end
+end
+
+# Set initial condition for t1
```

Comment:
@KiernanJ in the end of this tutorial, you have a function called `MHMPC`. Can you define this function earlier in the code? I think this is essentially just these next couple lines of code (136-137) that are wrapped in a function